src/imperative/imperative_utils.h

```diff
@@ -531,8 +541,18 @@ inline void PushOperator(const OpStatePtr& state,
         // copying A to B may not happen, and will corrupt A's memory.
         InvalidateOutputs(outputs, req);
       }
+      // add for mkldnn OP + no mkldnn OP
+      const auto is_mkldnn = Op::GetAttr<bool>("TIsMKLDNN");
+      if (!is_mkldnn.get(attrs.op, false)) {
+        std::vector<NDArray> inputs_fallback;
+        CreateDefaultInputs(inputs, &inputs_fallback);
+        fcompute_ex(state, opctx, inputs_fallback, req, outputs);
+      } else {
+#endif
+        fcompute_ex(state, opctx, inputs, req, outputs);
+#if MXNET_USE_MKLDNN == 1
+      }
```

Comment:
The macro condition check is a little bit complex here. Can we make it cleaner?

Reply:
OK. Changed in [16652](https://github.com/apache/incubator-mxnet/pull/16652)